### PR TITLE
Circular linked list

### PR DESCRIPTION
**Pull request by ghost**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implemented Circular Linked List in Golang

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#446

### CODE REVIEW

File: Go/ds/CircularLinkedList.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+// Returns the size of the given circular linked list
+func (list *circularLinkedList) length() int {
+	return list.size
+}
+
+func main() {
+	var list circularLinkedList
+
+	list.insert(0)
```
Comment (Kajol-Kumari) [inline Go/ds/CircularLinkedList.go]: Please take the input from user and provide the functionality in a `switch/case` statement.

File: Go/ds/Stack.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		fmt.Scan(&choice)
+
+		switch choice {
+		case 1:
+			element := getElement()
+			object.Push(element)
+		case 2:
+			element, err := object.Pop()
+			fmt.Println("Element: ", element)
+			fmt.Println("Error: ", err)
```
Comment (Kajol-Kumari) [inline Go/ds/Stack.go:82]: Can we please provide a check here? something like `if(err != null)` then only show it.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@odavhad Please incorporate the requested changes along with adding a sample input-output for both the program.

File: Go/ds/Stack.go (hunk 2 above)
Comment (ghost) [reply Go/ds/Stack.go:82]: okay, I'll do that.

**Pull request closed without merging**